sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DistinctHash.java

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation.query;
+
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.implementation.routing.MurmurHash3_128;
+import com.azure.cosmos.implementation.routing.UInt128;
+import com.azure.cosmos.models.ModelBridgeInternal;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+public class DistinctHash {
+
+    private static final UInt128 ArrayHashSeed = new UInt128(0xfa573b014c4dc18eL, 0xa014512c858eb115L);
+
+    private static final ObjectMapper OBJECT_MAPPER =
+        new ObjectMapper()
+            .configure(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY, true)
+            .configure(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS, true);
+
+    @SuppressWarnings("unchecked")
+    public static UInt128 getHash(Object resource) throws IOException {
+
+        if (resource instanceof List) {
+            return getHashFromList((List<Object>) resource);
+        }
+
+        if (resource instanceof JsonSerializable) {
+            return getHashFromJsonSerializable((JsonSerializable) resource);
+        }
+
+        final byte[] bytes = Utils.serializeObjectToByteArray(resource);
```

Comment:
This isn't going to work, since `1.0` and `1.00` need to hash to the same value. Also strings and their different escaped variants.

Reply:
Have tests for checking 1.0 vs 1